src/ray/raylet/scheduling/cluster_resource_manager.h

```diff
@@ -96,6 +96,26 @@ class ClusterResourceManager {
                              const ResourceRequest &resource_request,
                              bool ignore_object_store_memory_requirement) const;
 
+  /// Add available resource to a given node.
+  /// Return false if such node doesn't exist.
+  bool AddNodeAvailableResources(scheduling::NodeID node_id,
+                                 const ResourceRequest &resource_request);
+
+  /// Update node available resources.
+  /// NOTE: This method only updates the existing resources of the node, and the
+  /// nonexistent resources will be filtered out, whitch is different from `UpdateNode`.
+  /// Return false if such node doesn't exist.
+  bool UpdateNodeAvailableResources(scheduling::NodeID node_id,
+                                    const rpc::ResourcesData &resource_data);
+
+  /// Update node normal task resources.
+  /// Return false if such node doesn't exist.
+  bool UpdateNodeNormalTaskResources(scheduling::NodeID node_id,
+                                     const rpc::ResourcesData &resource_data);
+
+  /// Return false if the specified node doesn't exist.
+  bool ContainsNode(scheduling::NodeID node_id) const;
```

Comment:
Should this belong to the resource manager? Shouldn't this information be fetched somewhere else?

Reply:
> somewhere

Just as I wrote in the descriptions of this PR: Ideally, the `gcs_resource_manager` will be replaced with `cluster_resource_manager`, but now there are some PRC handlers and flush operations in the `gcs_resource_manager`, I just integrate `cluster_resource_manager` info `gcs_resource_manager` for the time being.

The `ContainsNode` method is only used inside `gcs_resource_manager` (UpdateResources, DeleteResources), which could be removed when `gcs_resource_manager` is replaced with `cluster_resource_manager`.